uxarray/grid/integrate.py

```diff
@@ -285,212 +456,227 @@ def _get_faces_constLat_intersection_info(
-        default_print_options = np.get_printoptions()
-        if (
-            str(e)
-            == "No intersections are found for the face, please make sure the build_latlon_box generates the correct results"
-        ):
-            # Set print options for full precision
-            np.set_printoptions(precision=16, suppress=False)
 
-            print(
-                "ValueError: No intersections are found for the face, please make sure the build_latlon_box generates the correct results"
-            )
-            print(f"Face edges information: {face_edges_cart}")
-            print(f"Constant z_0: {latitude_cart}")
-            print(f"Face latlon bound information: {face_latlon_bound}")
+@njit(cache=True, parallel=True)
+def _get_non_conservative_zonal_face_weights_at_const_lat_numba(
+    face_edges_xyz: np.ndarray,
+    n_edges_per_face: np.ndarray,
+    z: float,
+) -> np.ndarray:
+    """
+    Calculate normalized weights for faces intersecting a constant latitude using Numba
 
-            # Reset print options to default
-            np.set_printoptions(**default_print_options)
+    Parameters
+    ----------
+    face_edges_xyz : np.ndarray
+        Array of shape (n_face, max_edges, 2, 3) containing face edge coordinates
+    n_edges_per_face : np.ndarray
+        Array of shape (n_face,) containing the number of edges for each face
+    z : float
+        Z-coordinate of the constant latitude line
 
-            raise
-        else:
-            # Set print options for full precision
-            np.set_printoptions(precision=17, suppress=False)
+    Returns
+    -------
+    np.ndarray
+        Array of shape (n_face,) containing normalized weights for each face
+    """
+    n_face = face_edges_xyz.shape[0]
+    arc_lengths = np.zeros(n_face, dtype=np.float64)
 
-            print(f"Face edges information: {face_edges_cart}")
-            print(f"Constant z_0: {latitude_cart}")
-            print(f"Face latlon bound information: {face_latlon_bound}")
+    # 1) Pole Case: evenly distribute weights
+    if np.isclose(z, 1.0, atol=ERROR_TOLERANCE) or np.isclose(
+        z, -1.0, atol=ERROR_TOLERANCE
+    ):
+        return np.ones(n_face, dtype=np.float64) / n_face
 
-            # Reset print options to default
-            np.set_printoptions(**default_print_options)
+    # 2) Regular Case
+    for face_idx in prange(n_face):
+        n_edge = n_edges_per_face[face_idx]
+        face_data = face_edges_xyz[face_idx, :n_edge]  # shape (n_e, 2, 3)
+        arc_lengths[face_idx] = compute_face_arc_length(face_data, z)
 
-            raise  # Re-raise the exception if it's not the expected ValueError
+    total_arc = np.sum(arc_lengths)
+    return arc_lengths / total_arc
 
 
-def _process_overlapped_intervals(intervals_df):
-    """Process the overlapped intervals using the sweep line algorithm,
-    considering multiple intervals per face.
+def get_non_conservative_zonal_face_weights_at_const_lat(
```

Comment:
Similar naming and API-related comments as the `get_non_conservative_zonal_face_weights_at_const_lat_original ` function.